files/image_config/procdockerstats/procdockerstats

```diff
@@ -0,0 +1,212 @@
+# !/usr/bin/env python
+'''
+procdockerstats
+Daemon which periodically gathers process and docker statistics and pushes the data to STATE_DB
+'''
+
+import sys
+import os
+import time
+import syslog
+import subprocess
+import re
+import swsssdk 
+from datetime import datetime
+
+VERSION = '1.0'
+
+SYSLOG_IDENTIFIER = "procdockerstats"
+
+REDIS_HOSTIP = "127.0.0.1"
+
+# ========================== Syslog wrappers ==========================
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+# ========================== ProcessDocker class ==========================
+class ProcDockerStats:
+
+    def __init__(self):
+        self.state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+        self.state_db.connect("STATE_DB") 
+    
+    def run_command(self, cmd):
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        (stdout, stderr) = proc.communicate()
+        if proc.returncode != 0:
+            log_error("Error running command '{}'".format(cmd))
+            return None
+        else:
+            return stdout
+
+    def format_docker_cmd_output(self, cmdout):
+        lines = re.split("\n", cmdout)
+        keys = re.split("   +", lines[0])
+        docker_data = dict()
+        docker_data_list = []
+        for item in lines[1:]:
+            values1 = re.split("   +", item)
+            docker_data =  dict(zip(keys, values1))
+            docker_data_list.append(docker_data)
+        formatted_dict = self.create_docker_dict(docker_data_list)
+        return formatted_dict
+
+    def format_process_cmd_output(self, cmdout):
+        lines = re.split("\n", cmdout)
+        keys = re.split(" +", lines[0])
+        keylist = list(filter(None, keys))
+        process_data = dict()
+        process_data_list = []
+        for item in lines[1:]:
+            values1 = re.split(" +", str(item))
+            # To remove extra space before UID
+            val = list(filter(None, values1))
+            # Merging extra columns created due to space in cmd ouput
+            val[8:] = [''.join(val[8:])] 
+            process_data = dict(zip(keylist, val))
+            process_data_list.append(process_data)
+        return process_data_list
+
+    def convert_to_bytes(self, value):
+        unit_value = re.search('[a-zA-Z]+', value)
+        value_to_convert = float(filter(str.isdigit, value))
+        unit = unit_value.group(0)
+        UNITS_B = 'B'
+        UNITS_KB = 'KB'
+        UNITS_MB = 'MB'
+        UNITS_MiB = 'MiB'
+        UNITS_GiB = 'GiB'
+        if unit.lower() == UNITS_B.lower():
```

Comment:
Just curious why the need to convert both to lowercase before compare. Are you expecting case to differ from the constants above?

Reply:
Just to avoid any future change if happens